src/transformers/models/layoutlmv3/tokenization_layoutlmv3.py

```diff
@@ -517,7 +517,13 @@ def create_token_type_ids_from_sequences(
 
     def prepare_for_tokenization(self, text, is_split_into_words=False, **kwargs):
         add_prefix_space = kwargs.pop("add_prefix_space", self.add_prefix_space)
-        if (is_split_into_words or add_prefix_space) and (len(text) > 0 and not text[0].isspace()):
+        # If the text starts with a token that should not be split, no space is added before the text in any case.
+        # It's necessary to match the fast tokenization
+        if (
+            (is_split_into_words or add_prefix_space)
+            and (len(text) > 0 and not text[0].isspace())
+            and sum([text.startswith(no_split_token) for no_split_token in self.unique_no_split_tokens]) == 0
```

Comment:
I have to add this check if we want to have identical tokenization between slow and fast tokenizers in the case where our text is a space followed by an added token.

I've added a test to check that this is the case.